### ISSUE TITLE
principal_id_tag not being applied to S3 correctly with assume role

### ISSUE THREAD

**Issue by rheak** [cloud-custodian/cloud-custodian#3691]
When I apply tags using the policy below to an S3 bucket that was created by a user assuming a role, the principalId tag is not the full principalID tag of the user, it is the 1st half of the ID which is just the roleID only.

- name: s3-configure-standards-real-time
    resource: s3
    description: |
      This policy is triggered when a new S3 bucket is created and it applies
      the AWS AES256 Default Bucket Encryption, Tags the creators ID, enables
      object versioning, configures the bucket lifecycle and enables logging.
    mode:
      type: cloudtrail
      events:
        - CreateBucket
      role: arn:aws-us-gov:iam::12345678901:role/xxxxxxxx
      timeout: 200
    actions:
      - type: auto-tag-user
        tag: Owner
        principal_id_tag: PrincipalID
      - type: set-bucket-encryption
      - type: toggle-versioning
        enabled: true

![image](https://user-images.githubusercontent.com/18245244/54533069-f77dbc00-494e-11e9-82af-e760d3917d84.png)

Here is the first part of the cloudtrail event for the bucket creation that has the user info:
`{
    "eventVersion": "1.05",
    "userIdentity": {
        "type": "AssumedRole",
        "principalId": "AROAKxxxxWBRO1234:user.test",
        "arn": "arn:aws-us-gov:sts::12345678901:assumed-role/User-Role/user.test",
        "accountId": "12345678901",
        "accessKeyId": "QWERTYUIOPASDFGGHJKL",
        "sessionContext": {
            "attributes": {
                "mfaAuthenticated": "false",
                "creationDate": "2019-03-14T15:29:37Z"
            },
            "sessionIssuer": {
                "type": "Role",
                "principalId": "AROAKxxxxWBRO1234",
                "arn": "arn:aws-us-gov:iam::842617432181:role/WSO-CloudAdmin",
                "accountId": "12345678901",
                "userName": "User-Role"
            }
        }
    },`

**Comment by kapilt** [comment]
what outcome are you expecting? are you saying want the same info in duplicated in two different tags?

**Comment by rheak** [comment]
I think the user principal Identity should be 'RoleID:userID' ("principalId": "AROAKxxxxWBRO1234:user.test"). It seems like it is just pulling the Role Principal Identity ("principalId": "AROAKxxxxWBRO1234") 

For my use case, Im building policies that depend on the principalID being the full principal Identity of the user. (https://aws.amazon.com/blogs/security/how-to-restrict-amazon-s3-bucket-access-to-a-specific-iam-role/)

The AWS Variable userID is = to the full principalID of the user (RoleID:UserID).

**Comment by kapilt** [comment]
something worth keeping in mind here especially if your building iam policies based on these values, assume-role is not just roleid:userid, its actually session name, on saml/sso identity this may be well defined, for assumed-role its whatever string a user passed in for session name on sts.AssumeRole api call.

does the conjunction of principal_id_tag and owner_tag give the value your looking for? wrt to directly changing this, we'll need to keep compatibiltiy concerns in mind. An alternative would be let a user define a format string for the value, but the variables / values that are available will be dependent on form of authorization/usertype, tbd.

**Comment by rheak** [comment]
Yep I agree. Here is the AWS Doc that shows what the expected value should be for the variables determined by authenticated session type. https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html

Yes the conjunction of principal_id_tag and owner_tag give me the value Im looking for. Is it possible to concatenate those values somehow? (forgive me if that is a dumb question still newer to Custodian)

**Comment by konturn** [comment]
While I didn't fix the `principal_id_tag` to reflect the exact value of the principalId (for compatibility reasons), I did add the option to use the newer `value` field to pull out the exact value of the principalId. This can probably be closed.
